src/Frontend/TableHelper/Animation.js

```diff
@@ -24,227 +19,258 @@ export const stopStatus = { // 暫停時的狀態
     for (var i = 0; i < search.length; i++) {
         for (var j = 0; j < search[i].length; j++) {
-            if (WhichComponentSame(search[i][j]) > 3) {
-                if(position.bomb === false){
-                    setTable(search[i][j], componentKind.searchStatic)
-                }else{
-                    setTable(search[i][j], componentKind.searchBombStatic)
-                }
-            }else{
-                if(position.bomb === false){
-                    setTable(search[i][j], StartEndBombWeight(WhichComponentSame(search[i][j]), componentKind.startSearch, componentKind.endSearch, componentKind.bombSearch, componentKind.weightSearchStatic))
-                }else{
-                    setTable(search[i][j], StartEndBombWeight(WhichComponentSame(search[i][j]), componentKind.startSearchBomb, componentKind.endSearchBomb, componentKind.bombSearchBomb, componentKind.weightSearchBombStatic))
-                }
-            }
+            id = search[i][j]; type = WhichComponentSame(id);
+            className1 = (type > 3) ? componentKind.searchStatic : StartEndBombWeight(type, componentKind.startSearch, componentKind.endSearch, componentKind.bombSearch, componentKind.weightSearchStatic);
+            className2 = (type > 3) ? componentKind.searchBombStatic : StartEndBombWeight(type, componentKind.startSearchBomb, componentKind.endSearchBomb, componentKind.bombSearchBomb, componentKind.weightSearchBombStatic);
+            (position.bomb === false) ? setTable(id, className1) : setTable(id, className2);
         }
     }
     for (i = 0; i < bomb.length; i++) {
         for (j = 0; j < bomb[i].length; j++) {
-            if (WhichComponentSame(bomb[i][j]) > 3) {
-                setTable(bomb[i][j], componentKind.searchStatic)
-            }else{
-                setTable(bomb[i][j], StartEndBombWeight(WhichComponentSame(bomb[i][j]), componentKind.startSearch, componentKind.endSearch, componentKind.bombSearch, componentKind.weightSearchStatic))
-            }
+            id = bomb[i][j]; type = WhichComponentSame(id);
+            className1 = (type > 3) ? componentKind.searchStatic : StartEndBombWeight(WhichComponentSame(id), componentKind.startSearch, componentKind.endSearch, componentKind.bombSearch, componentKind.weightSearchStatic);
+            setTable(id, className1);
         }
     }
     for (i = 0; i < path.length; i++) {
-        if (WhichComponentSame(path[i]) > 3) {
-            setTable(path[i], componentKind.pathStatic)
-        }else{
-            setTable(path[i], StartEndBombWeight(WhichComponentSame(path[i]), direction(pathDirection[i]), direction(pathDirection[i]), componentKind.bombPath, componentKind.weightPathStatic))
-        }
+        id = path[i]; d = pathDirection[i]; type = WhichComponentSame(id);
+        className1 = (type > 3) ? componentKind.pathStatic : StartEndBombWeight(WhichComponentSame(id), direction(d), direction(d), componentKind.bombPath, componentKind.weightPathStatic);
```

Comment:
Please write a function named "isWall" for "(type > 3)".
Make sure you change all the references.

Reply:
type > 3 => isWall || isPath